### PR TITLE
Take various updates from `fez` repository

### PR DESCRIPTION
**Pull request by cmeister2**
Here's the updates from `fez`. I suggest we rename `master` to `main` as well.

I'd love it if my pet feature made it in (fast RPM reading) but I've not included in this MR.

---

`fez` was a fork of `rpm-rs` with a number of maintenance fixes. This
commit takes those maintenance fixes and ports them into `rpm-rs`

### CODE REVIEW

**Comment by Shinyzenith** [comment]
PR looks good apart from the failing tests :D tysm cmeister!!

**Review by Shinyzenith — DISMISSED** [review]

**Comment by Shinyzenith** [comment]
Also one minor nitpick: If the commit message could be a bit more descriptive it would help later on since not everyone knows about fez project

**Comment by cmeister2** [comment]
> Also one minor nitpick: If the commit message could be a bit more descriptive it would help later on since not everyone knows about fez project

I've updated it with a little more detail.

**Review by Shinyzenith — DISMISSED** [review]

**Review by drahnr — CHANGES_REQUESTED** [review]
nit: there is a stray TODO that should be moved to an issue and be references, otherwise LGTM!

**Comment by cmeister2** [comment]
> mit: there is a stray TODO that should be moved to an issue and be references, otherwise LGTM!

Fixed.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
+updates:
+- package-ecosystem: cargo
+  directory: "/"
+  schedule:
+    interval: daily
```
Comment (drahnr) [inline .github/dependabot.yml:6]: ``
```suggestion
    interval: weekly
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,161 @@
+on:
+  push:
+    tags:
+      # Full version
+      - "[0-9]+.[0-9]+.[0-9]+"
```
Comment (drahnr) [inline .github/workflows/ci.yml:5]: nit: prefixing with `v` might be a good idea to create a common prefix - it also seems to be a common pattern

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      # Prerelease version
+      - "[0-9]+.[0-9]+.[0-9]+-*"
+
+  pull_request:
+    branches:
+      # Trigger on pull requests into main/master
+      - main
```
Comment (drahnr) [inline .github/workflows/ci.yml:12]: I'll take care of the move to main after the merge

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+jobs:
+  check:
+    name: Check
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - 1.57.0
```
Comment (drahnr) [inline .github/workflows/ci.yml:26]: Does this align with the cargo manifest msrv? If not, that should be one a follow up PR, of which I can take care if

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        with:
+          command: install
+          args: cargo-tarpaulin
+      - uses: actions-rs/cargo@v1
+        with:
+          command: tarpaulin
+          args: --out xml
+      # Commented out until a codecov token is added
+
+      # - uses: codecov/codecov-action@v3
```
Comment (drahnr) [inline .github/workflows/ci.yml:123]: I'd prefer to delete commented code and introduce functional code all at once. We should move this to a separate PR. Do you want to take care of getting a token or should I?

File: src/rpm/signature/pgp.rs [hunk 6]
```diff
@@ -31,7 +31,7 @@ impl traits::Signing<traits::algorithm::RSA> for Signer {
     /// it internally creates a copy until crate `pgp` provides
     /// a `Read` based implementation.
     fn sign<R: Read>(&self, data: R) -> Result<Self::Signature, RPMError> {
-        let passwd_fn = || String::new();
+        let passwd_fn = String::new;
```
Comment (drahnr) [inline src/rpm/signature/pgp.rs:34]: 👍

File: .github/workflows/ci.yml (hunk 2 above)
Comment (cmeister2) [reply .github/workflows/ci.yml:5]: Generally speaking I'm not a fan of a v prefix. Having the tag match the Cargo version (and also be semver) tends to be quite useful.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (cmeister2) [reply .github/workflows/ci.yml:26]: It ought to - I'll double check when I'm back at a PC

File: .github/workflows/ci.yml (hunk 5 above)
Comment (cmeister2) [reply .github/workflows/ci.yml:123]: I don't mind. I can sort it out assuming I have access to do so.

Happy to delete commented-out code if you want - felt it was probably acceptable in the CI files temporarily.

File: .github/dependabot.yml (hunk 1 above)
Comment (cmeister2) [reply .github/dependabot.yml:6]: Don't mind.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (cmeister2) [reply .github/workflows/ci.yml:26]: According to `cargo msrv` the actual MSRV is 1.56.1, so we should probably run with that.

File: .github/workflows/ci.yml (hunk 5 above)
Comment (cmeister2) [reply .github/workflows/ci.yml:123]: Have requested access for adding codecov. But I don't have access to add secrets to the repository.

Comment (drahnr) [reply .github/workflows/ci.yml:123]: Added you. I'll grant access based on contributions over the next few weeks, so we have a few admins :)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (cmeister2) [reply .github/workflows/ci.yml:5]: I've added in the v variants so we have a choice.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (drahnr) [reply .github/workflows/ci.yml]: And also tie it down in the manifest

**Review by drahnr — DISMISSED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (cmeister2) [reply .github/workflows/ci.yml]: Ignore that, `cargo msrv` is blatantly lying 😂. I'll revert and make it 1.57.0 again.

```
`time v0.3.11` cannot be built because it requires rustc 1.57.0 or newer, while the currently active rustc version is 1.56.1
```

**Comment by cmeister2** [comment]
@drahnr @Shinyzenith just need your reapproval whenever is convenient!

**Review by drahnr — APPROVED** [review]

**Review by Shinyzenith — APPROVED** [review]
